src/utilities/exposed-api-key-manager.ts

```diff
@@ -0,0 +1,62 @@
+import * as vscode from "vscode";
+
+import { SecretApiKeyManager } from "./secret-storage-manager";
+import { error } from "console";
+
+export class ExposedApiKeyManager {
+    readonly context: vscode.ExtensionContext;
+
+    constructor(context: vscode.ExtensionContext) {
+        this.context = context;
+    }
+
+    public async checkAndShiftConfigApiKey() {
+        //check if api key is in config then take the api key and shift to secretStorage
+        let apiKey: string | undefined =
+            await this.getApiKeyFromConfig();
+            let isApiKeyShifted = false;
+        if (apiKey) {
+            //SET API KEY IN SECRET STORAGE
+           isApiKeyShifted =  await  SecretApiKeyManager.instance.setApiKey(
+                
+                apiKey
+            );
+            
+        }
+        // we delete the key from config only if we successfully shift the key to secret storage
+        if(isApiKeyShifted){
+this.deleteApiKeyFromConfig();
```

Comment:
add a telemetry event please that marks successful shift.

Reply:
cool will add it